component/portal/src/main/java/org/exoplatform/portal/rest/UserRestResourcesV1.java

```diff
@@ -528,7 +528,7 @@ public Response isSynchronizedUserAllowedToChangePassword(@Context UriInfo uriIn
     if (user == null) {
       throw new WebApplicationException(Response.Status.BAD_REQUEST);
     }
-    boolean allowChangePassword = Boolean.valueOf(System.getProperty("exo.portal.allow.change.external.password").toString()) || user.isInternalStore();
+    boolean allowChangePassword = (System.getProperty("exo.portal.allow.change.external.password") != null && Boolean.valueOf(System.getProperty("exo.portal.allow.change.external.password").toString())) || user.isInternalStore();
```

Comment:
```suggestion
    boolean allowChangePassword = user.isInternalStore() || (System.getProperty("exo.portal.allow.change.external.password") != null && Boolean.valueOf(System.getProperty("exo.portal.allow.change.external.password").toString()));
```

In fact, the case we will see in majority is the user in internal store, so it is preferable to check it first